apps/extension/server/src/server.ts

```diff
@@ -34,9 +36,11 @@ import { getCursorHoverType } from './getCursorHoverType';
 import { getDiagnostics } from './getDiagnostics';
 import { getReferences } from './getReferences';
 import { CachedDocument } from './types';
+import { isErrorWithMessage } from './utils';
 
 let hasConfigurationCapability = false;
 let hasWorkspaceFolderCapability = false;
+let hasActiveError = false;
```

Comment:
Because this variable is "global" right now it can potentially lead to bugs down the road. I think that right now it won't necessarily lead to a bug but there is a data inconsistency problem here and, in the future, someone could accidentally use this incorrectly and thus introduce an actual bug without knowing it.

The problem is that the user can switch to displaying a different machine through a code lens. This requests the machine's data from the server and forwards that to the editor using `UPDATE_CONFIG` event. Based on that we already clear the error in the editor so it won't be displayed there but the `hasActiveError` variable here won't be cleaned up. Note that such a scenario isn't related to the document change and thus `handleDocumentChange` won't be called and won't reset the `hasActiveError` to `false` just yet.

For that reason, I would track those errors more locally - per machine - in the `documentsCache`. Syntax errors have to be added to all machines in the file but config-level errors should only be added to a single machine.

Reply:
And actually, there is a bug here now - because config-level errors are global now, we might introduce a config-related error to a machine that is not displayed and yet we'd display the error there. This should be avoided.

We can't as easily rely on a catch-all `catch` for those config-related errors. We should actually catch them per `introspectMachine` call while mapping this. Since `getTypegenData` also calls that `introspectMachine` under the hood - it would probably be best to map machines in a loop to an array close to what I've proposed [here](https://github.com/statelyai/xstate-tools/pull/242#discussion_r1020237668):
> We need to end up with an array like this Array<{ types?: TypegenData, configError?: string }> and handle that appropriately in logic that follows this .map.

Reply:
Thanks for the input; it makes sense to handle this more locally. I've tried to do a version in the latest commit; let me know what you think.